### PR TITLE
[BACKLOG-22046] - Created a proxy that maps between internal and external networks to expose the Solr service when Foundry running with multiple instances

### PR DESCRIPTION
**Pull request by RPAraujo**
@webdetails/millenniumfalcon please review

### CODE REVIEW

File: assemblies/solr-proxy-service/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<features name="${project.artifactId}-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.0.0">
+
+    <feature name='Solr Proxy Service' description='Solr Proxy Service' version='${project.version}'>
+        <bundle>mvn:org.pentaho.ctools/solr-proxy/${project.version}</bundle>
```
Comment (nantunes) [inline assemblies/solr-proxy-service/src/main/feature/feature.xml]: The bundle should be added as dependency in the pom file and inserted in the feature by the karaf-maven-plugin plugin. But we probably don't need the feature anyways, the bundle is enough.

File: pom.xml [hunk 2]
```diff
@@ -22,6 +22,7 @@
     <module>components</module>
     <module>webclient</module>
     <module>assemblies</module>
+    <module>solr</module>
```
Comment (nantunes) [inline pom.xml]: We need to rethink the "position" of the module... The root of the repo seems too much for a temporary hack...

File: solr/solr-proxy/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>3.5.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Import-Package>!org.eclipse.jetty.servlets.ProxyServlet,org.eclipse.jetty
```
Comment (nantunes) [inline solr/solr-proxy/pom.xml]: Put a small comment above this line explaining why we need to explicitly exclude `org.eclipse.jetty.servlets.ProxyServlet` (incorrect handling by karaf-maven-plugin of inner classes use).

File: solr/pom.xml [hunk 4]
```diff
@@ -9,11 +9,39 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <!-- We need to explicitlly exclude org.eclipse.jetty.servlets.ProxyServlet
+                             because we are using the Transparent inner class and the plugin is using
+                             org.eclipse.jetty.servlets.ProxyServlet like a package name, witch is wrong. -->
+                        <Import-Package>!org.eclipse.jetty.servlets.ProxyServlet,org.eclipse.jetty
```
Comment (dcleao) [inline solr/pom.xml]: `witch` -> `which` :-D

Comment (RPAraujo) [reply solr/pom.xml]: :-D

File: assemblies/cde-renderer/feature/pom.xml [hunk 5]
```diff
@@ -12,6 +12,16 @@
   <artifactId>pentaho-cde-renderer</artifactId>
   <packaging>feature</packaging>
 
+  <dependencyManagement>
```
Comment (nantunes) [inline assemblies/cde-renderer/feature/pom.xml]: Why is this here?

File: assemblies/cde-renderer/solr/pom.xml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-servlets</artifactId>
+            <version>8.1.19.v20160209</version>
```
Comment (nantunes) [inline assemblies/cde-renderer/solr/pom.xml]: Externalize the version to a property.
